File: app/buildz/tmp/kapt3/stubs/debug/com/bits/bee/bpmc/data/repository/LoginRepositoryImpl.java
package com.bits.bee.bpmc.data.repository;

import java.lang.System;

/**
 * Created by aldi on 02/03/22.
 */
@kotlin.Metadata(mv = {1, 7, 1}, k = 1, d1 = {"\u0000(\n\u0002\u0018\u0002\n\u0002\u0018\u0002\n\u0000\n\u0002\u0018\u0002\n\u0002\b\u0002\n\u0002\u0018\u0002\n\u0002\u0018\u0002\n\u0002\u0018\u0002\n\u0000\n\u0002\u0010\u000e\n\u0002\b\u0002\u0018\u00002\u00020\u0001B\u000f\b\u0007\u0012\u0006\u0010\u0002\u001a\u00020\u0003\u00a2\u0006\u0002\u0010\u0004J$\u0010\u0005\u001a\u000e\u0012\n\u0012\b\u0012\u0004\u0012\u00020\b0\u00070\u00062\u0006\u0010\t\u001a\u00020\n2\u0006\u0010\u000b\u001a\u00020\nH\u0016R\u000e\u0010\u0002\u001a\u00020\u0003X\u0082\u0004\u00a2\u0006\u0002\n\u0000\u00a8\u0006\f"}, d2 = {"Lcom/bits/bee/bpmc/data/repository/LoginRepositoryImpl;", "Lcom/bits/bee/bpmc/domain/repository/LoginRepository;", "utils", "Lcom/bits/bee/bpmc/data/data_source/remote/ApiUtils;", "(Lcom/bits/bee/bpmc/data/data_source/remote/ApiUtils;)V", "login", "Lkotlinx/coroutines/flow/Flow;", "Lcom/bits/bee/bpmc/utils/Resource;", "Lcom/bits/bee/bpmc/data/data_source/remote/response/LoginResponse;", "email", "", "password", "app_debug"})
public final class LoginRepositoryImpl implements com.bits.bee.bpmc.domain.repository.LoginRepository {
    private final com.bits.bee.bpmc.data.data_source.remote.ApiUtils utils = null;
    
    @javax.inject.Inject()
    public LoginRepositoryImpl(@org.jetbrains.annotations.NotNull()
    com.bits.bee.bpmc.data.data_source.remote.ApiUtils utils) {
        super();
    }
    
    @org.jetbrains.annotations.NotNull()
    @java.lang.Override()
    public kotlinx.coroutines.flow.Flow<com.bits.bee.bpmc.utils.Resource<com.bits.bee.bpmc.data.data_source.remote.response.LoginResponse>> login(@org.jetbrains.annotations.NotNull()
    java.lang.String email, @org.jetbrains.annotations.NotNull()
    java.lang.String password) {
        return null;
    }
}

File: app/src/main/java/com/bits/bee/bpmc/data/repository/LoginRepositoryImpl.kt
package com.bits.bee.bpmc.data.repository

import com.bits.bee.bpmc.data.data_source.remote.ApiUtils
import com.bits.bee.bpmc.data.data_source.remote.model.DbPost
import com.bits.bee.bpmc.data.data_source.remote.post.LoginPost
import com.bits.bee.bpmc.data.data_source.remote.response.DbResponse
import com.bits.bee.bpmc.data.data_source.remote.response.LoginResponse
import com.bits.bee.bpmc.domain.repository.LoginRepository
import com.bits.bee.bpmc.utils.ApiResponse
import com.bits.bee.bpmc.utils.NetworkBoundResource
import com.bits.bee.bpmc.utils.Resource
import kotlinx.coroutines.flow.Flow
import javax.inject.Inject

/**
 * Created by aldi on 02/03/22.
 */

class LoginRepositoryImpl @Inject constructor(
    private val utils: ApiUtils
) : LoginRepository {

     override fun login(email: String, password: String): Flow<Resource<LoginResponse>> {
        val loginPost = LoginPost(email, password)
        return object : NetworkBoundResource<LoginResponse>(){
            override fun createCall(): Flow<ApiResponse<LoginResponse>> {
                return utils.getAuthApiService().login(loginPost)
            }
        }.getAsFlow()
    }

    override fun postDb(usename: String, dbName: String): Flow<Resource<DbResponse>> {
        return object : NetworkBoundResource<DbResponse>(){
            override fun createCall(): Flow<ApiResponse<DbResponse>> {
                return utils.getAuthApiService().postDB(DbPost(username = usename, dbName))
            }
        }.getAsFlow()
    }
}